toqito/matrix_props/spark.py

```diff
@@ -0,0 +1,47 @@
+"""Compute spark of matrix."""
+
+from itertools import combinations
+
+import numpy as np
+
+
+def spark(mat: np.ndarray) -> int:
+    """Compute the spark of a matrix.
+
+    The spark of a matrix A is the smallest number of columns from A that are linearly dependent [Elad_2010_Sparse].
+
+    Examples
+    =========
+    >>> import numpy as np
+    >>> from toqito.matrix_props import spark
+    >>> A = np.array([[1, 0, 1, 2],
+    ...               [0, 1, 1, 3],
+    ...               [1, 1, 2, 5]])
+    >>> spark(A)
+    3
+
+    Notes
+    =====
+    - If all columns are linearly independent, the function returns n_cols + 1.
+    - The time complexity of this implementation is O(2^n) in the worst case,
+      where n is the number of columns.
```

Comment:
```suggestion
    - If all columns are linearly independent, the function returns $n_{\text{cols}} + 1$.
    - The time complexity of this implementation is $O(2^n)$ in the worst case,
      where $n$ is the number of columns.
```

Reply:
I think you should add a linearly independent example in addition to adding this note.

Reply:
Fixed!